src/Provider/StandardProvider.php

```diff
@@ -0,0 +1,157 @@
+<?php
+
+namespace League\OAuth2\Client\Provider;
+
+use InvalidArgumentException;
+
+class StandardProvider extends AbstractProvider
+{
+    /**
+     * @var string
+     */
+    private $urlAuthorize;
```

Comment:
I don't usually like `private`, but I think it makes sense in this case... all the configuration is external and should not be modified after the fact. This keeps it as immutable as an extended class.